plan/plan.go

```diff
@@ -222,7 +224,8 @@ func (p *Plan) Calculate() *Plan {
 				if records.current != nil && len(records.candidates) > 0 {
 					update := t.resolver.ResolveUpdate(records.current, records.candidates)
 
-					if shouldUpdateTTL(update, records.current) || targetChanged(update, records.current) || p.shouldUpdateProviderSpecific(update, records.current) {
+					if shouldUpdateTTL(update, records.current) || targetChanged(update, records.current) || p.shouldUpdateProviderSpecific(update, records.current) ||
+						len(p.OldOwnerId) != 0 && records.current.Labels[endpoint.OwnerLabelKey] != p.OldOwnerId {
```

Comment:
can we have this as a method as well?

Reply:
Missing unit test for added code block

I've added dummy `if` statement to validate that

![Screenshot 2025-06-23 at 06 41 52](https://github.com/user-attachments/assets/088bf3cd-c5c2-4a6c-a2ce-8e8cb8d18cb1)